models/opnilog/opnilog_parser.py

```diff
@@ -366,9 +402,20 @@ def run_epoch(self, dataloader, model, loss_compute):
 
             if i % self.step_size == 1:
                 elapsed = time.time() - start
+                training_progress = ((i / len(dataloader)) + epoch) / self.nr_epochs
+                total_time_taken = time.time() - training_start_time
+                remaining_time = (
+                    total_time_taken // training_progress
+                ) - total_time_taken
                 logging.info(
-                    "Epoch Step: %d / %d Loss: %f Tokens per Sec: %f"
-                    % (i, len(dataloader), loss / batch.ntokens, tokens / elapsed)
+                    f"| Epoch: {epoch} | Total Progress: {(training_progress * 100):.2f}% | Training Time Taken: {total_time_taken:.2f}s | ETC: {(remaining_time):.2f}s | Epoch Step: {i}/{len(dataloader)} | Loss: {(loss / batch.ntokens):.4f} | Tokens per Sec: {(tokens / elapsed):.2f} |"
+                )
+                post_model_stats(
```

Comment:
can we make this optional , and disabled by default?

Reply:
Done